auth_token/enums.py

```diff
@@ -0,0 +1,25 @@
+from django.utils.translation import ugettext_lazy as _
+
+from enumfields import Choice, ChoiceEnum
+
+
+class AuthorizationRequestType(ChoiceEnum):
+
+    OTP = Choice(1, _('OTP'))
+    MOBILE_DEVICE = Choice(2, _('mobile device'))
+
+
+class AuthorizationRequestResult(ChoiceEnum):
+
+    GRANTED = Choice(1, _('granted'))
+    DENIED = Choice(2, _('denied'))
+    CANCELLED = Choice(3, _('cancelled'))
+
+
+class AuthorizationRequestState(ChoiceEnum):
+
+    GRANTED = Choice(1, _('granted'))
+    DENIED = Choice(2, _('denied'))
+    CANCELLED = Choice(3, _('cancelled'))
```

Comment:
Because this is two different states. One is stored and the second one computed. Yes I can use AuthorizationRequestState as AuthorizationRequestResult but I have no way how to restrict model not to save WAITING and EXPIRED state (yes I can write clean by hand, but it is ugly)